src/libs/relay/conduit_relay_io_silo.cpp

```diff
@@ -3789,100 +3777,111 @@ void silo_write_ucd_zonelist(DBfile *dbfile,
     }
     else
     {
-        n_mesh_conn.set_external(n_elements["connectivity"]);
+        n_conn.set_external(n_elements["connectivity"]);
     }
 
-    if (topo_shape == "quad")
+    int total_num_elems = 0;
+
+    std::vector<int> shapetype;
+    std::vector<int> shapesize;
+    std::vector<int> shapecnt;
+
+    index_t num_shapes = 0;
+
+    auto set_up_single_shape_type = [&](index_t num_pts, int db_zonetype)
     {
-        int num_elems = n_mesh_conn.dtype().number_of_elements() / 4;
-        shapetype[0] = DB_ZONETYPE_QUAD;
-        shapesize[0] = 4;
-        shapecnt[0] = num_elems;
+        num_shapes = 1;
+
+        int num_elems = n_conn.dtype().number_of_elements() / num_pts;
+
+        shapetype.push_back(db_zonetype);
+        shapesize.push_back(num_pts);
+        shapecnt.push_back(num_elems);
+
         total_num_elems += num_elems;
+    };
+
+    // TODO should we allow people to write tris, wedges, and pyramids to overlink?
+    // I don't think overlink supports those shape types
+    // but we still produce valid silo
```

Comment:
What is the right call here?

Reply:
if the overlink validator would fail, we should raise an error b/c it;s not inline with overlink std.

In the the error message we can suggest they can use generic silo for this case

Reply:
Ok sounds good.